### ISSUE TITLE
Folding/unfolding happens wherever the mouse is clicked in the folding column 

### ISSUE THREAD

**Issue by shetlandbreeder** [geany/geany#2266]
At the moment (Geany 1.35 on Arch Linux) clicking anywhere in the fold column folds that section.

This seems like a bug (at least it's _really_ annoying) as the documentation says folding only happens if you click on the fold icon.

If it's not a bug then this behaviour should be documented and ideally be configurable.

**Comment by ntrel** [comment]
It's intentional. I assume it's a problem for you due to accidental clicks?

**Comment by shetlandbreeder** [comment]
Yes, it's accidental clicks. I've been using Scite for many years and it doesn't do this, so when selecting a line I'm used to being able to click anywhere in the line number or folds bars.

I switched because  lot of the syntax checkers on Scite are very out of date - CSS in particular. Mybe I should just bite the bullet and update them if this behaviour is not configurable .

**Comment by elextr** [comment]
Yes its deliberate, and no it has no configuration, see [here](https://github.com/geany/geany/blob/d80e36e85a031fc266fe200add05f0626a0696ad/src/editor.c#L465).

Pull requests are welcome.

**Comment by ntrel** [comment]
It should probably be a Various pref rather than a dialog option IMO.

**Comment by surruk51** [comment]
This is really annoying me too. I don't think I have the skill to patch it, so if someone can do this I would be eternally grateful. Not only does it fold, but when you unfold, the previous cursor position seems to be lost. 
Suggested configurable behaviour: fold_on = click | dblclick | none
